File: src/main/java/saheed/samplejavacpp/MathApp.java
package saheed.samplejavacpp;


import org.bytedeco.javacpp.*;
import org.bytedeco.javacpp.annotation.*;


/*
 * Place code in accessible place
 * Add references and imports
 * Include the path to the code
 * declare native methods
 * Call Loader.load() statically
 */


@Platform(include="resources/MathApp1/MathApp.cpp")
public class MathApp {
	
	native int add(int a, int b);
    native int subtract(int a, int b);
    native int multiply(int a, int b);
    
    static {
        Loader.load();
    }

	
}

File: src/main/resources/MathApp1/MathApp.cpp
#include <iostream>

int add(int a, int b){return a+b;}
int subtract(int a, int b){return a-b;}
int multiply(int a, int b){return a*b;}

using namespace std;

int main(){
    int a = 19, b = 23;
    cout<<"\n\n";
    cout<< a<< " + " << b << " = " << add(a,b) <<"\n"; 
    cout<< a<< " - " << b << " = " << subtract(a,b) <<"\n"; 
    cout<< a<< " * " << b << " = " << multiply(a,b) <<"\n"; 
}